### ISSUE TITLE
loss NAN

### ISSUE THREAD

**Issue by RichieMay** [sirius-ai/MobileFaceNet_TF#21]
我用softmax训练17万次后（lr=0.1, batch 90）再用arcface训练[batch 90]几万步就会出现loss 无效的问题，请问这个要怎么解决？大神有完整的训练过吗，从softmax 到arcface 参数设置?

**Comment by rimio** [comment]
See issue #24 .

**Comment by jetsmith** [comment]
@rimio @RichieMay i got the similar problem.  In my experiment, batch size is set to 64 and i use single GTX1080Ti.
 first i train softmax but the training acc is always 0, here is the training log: 
epoch 5, total_step 349550, total loss is 6.06 , inference loss is 5.59, reg_loss is 0.48, training accuracy is 0.000000, time 283.005 samples/sec
epoch 5, total_step 349600, total loss is 11.22 , inference loss is 10.74, reg_loss is 0.48, training accuracy is 0.000000, time 271.270 samples/sec
epoch 5, total_step 349650, total loss is 8.83 , inference loss is 8.36, reg_loss is 0.48, training accuracy is 0.000000, time 278.559 samples/sec
epoch 5, total_step 349700, total loss is 5.12 , inference loss is 4.64, reg_loss is 0.48, training accuracy is 0.000000, time 282.140 samples/sec
epoch 5, total_step 349750, total loss is 6.47 , inference loss is 6.00, reg_loss is 0.48, training accuracy is 0.000000, time 285.530 samples/sec
epoch 5, total_step 349800, total loss is 6.01 , inference loss is 5.53, reg_loss is 0.48, training accuracy is 0.000000, time 275.156 samples/sec
epoch 5, total_step 349850, total loss is 11.20 , inference loss is 10.72, reg_loss is 0.48, training accuracy is 0.000000, time 281.050 samples/sec
epoch 5, total_step 349900, total loss is 8.33 , inference loss is 7.85, reg_loss is 0.48, training accuracy is 0.000000, time 280.594 samples/sec
epoch 5, total_step 349950, total loss is 8.60 , inference loss is 8.12, reg_loss is 0.48, training accuracy is 0.000000, time 278.497 samples/sec
epoch 5, total_step 350000, total loss is 6.44 , inference loss is 5.97, reg_loss is 0.48, training accuracy is 0.000000, time 292.478 samples/sec

Iteration 350000 testing...
thresholds max: 1.71 <=> min: 0.3
total time 16.082s to evaluate 12000 images of lfw
Accuracy: 0.691+-0.015
Validation rate: 0.09200+-0.01881 @ FAR=0.00167
fpr and tpr: 0.573 0.782
Area Under Curve (AUC): 0.761
Equal Error Rate (EER): 0.322

Then, i try to train arcface directly without pretraining and all settings are the same as souce code's, here is arcface log and NaN happened:
epoch 0, total_step 2200, total loss is 19.51 , inference loss is 18.61, reg_loss is 0.90, training accuracy is 0.000000, time 6.838 samples/sec
epoch 0, total_step 2250, total loss is 19.84 , inference loss is 18.94, reg_loss is 0.90, training accuracy is 0.000000, time 3.714 samples/sec
epoch 0, total_step 2300, total loss is 18.70 , inference loss is 17.81, reg_loss is 0.90, training accuracy is 0.000000, time 6.292 samples/sec
epoch 0, total_step 2350, total loss is 18.68 , inference loss is 17.79, reg_loss is 0.89, training accuracy is 0.000000, time 6.409 samples/sec
epoch 0, total_step 2400, total loss is 20.18 , inference loss is 19.30, reg_loss is 0.89, training accuracy is 0.000000, time 6.463 samples/sec
epoch 0, total_step 2450, total loss is 18.90 , inference loss is 18.02, reg_loss is 0.88, training accuracy is 0.000000, time 6.212 samples/sec
epoch 0, total_step 2500, total loss is nan , inference loss is nan, reg_loss is nan, training accuracy is 0.000000, time 6.386 samples/sec
epoch 0, total_step 2550, total loss is nan , inference loss is nan, reg_loss is nan, training accuracy is 0.000000, time 6.554 samples/sec
epoch 0, total_step 2600, total loss is nan , inference loss is nan, reg_loss is nan, training accuracy is 0.000000, time 2.952 samples/sec
epoch 0, total_step 2650, total loss is nan , inference loss is nan, reg_loss is nan, training accuracy is 0.000000, time 6.242 samples/sec
epoch 0, total_step 2700, total loss is nan , inference loss is nan, reg_loss is nan, training accuracy is 0.000000, time 6.493 samples/sec
epoch 0, total_step 2750, total loss is nan , inference loss is nan, reg_loss is nan, training accuracy is 0.000000, time 6.529 samples/sec
epoch 0, total_step 2800, total loss is nan , inference loss is nan, reg_loss is nan, training accuracy is 0.000000, time 6.532 samples/sec
So how can i solve it .Please help me  thanks

**Comment by onepre** [comment]
@jetsmith   you got training accuracy is 0.000000........but,   Iteration 350000 testing...,,  you got test data as below:, i feel so confused about the result   accuracy is 0.691   ??   could you pls. tell me  why that would happen ?  thank you !

Accuracy: 0.691+-0.015
Validation rate: 0.09200+-0.01881 @ FAR=0.00167
fpr and tpr: 0.573 0.782
Area Under Curve (AUC): 0.761
Equal Error Rate (EER): 0.322

**Comment by sirius-ai** [comment]
close this issue, when need reopen.

**Comment by zfs1993** [comment]
> 我用softmax训练17万次后（lr=0.1, batch 90）再用arcface训练[batch 90]几万步就会出现loss 无效的问题，请问这个要怎么解决？大神有完整的训练过吗，从softmax 到arcface 参数设置?

我有同样的问题，但是我直接从arcface训练的，请问你后来解决了吗？
